docs/source/opkg-keyrings/opkg-keyrings_index.rst

```diff
@@ -0,0 +1,157 @@
+    * The public key is distributed to the target systems.
+
+#. Sign the Package Index:
+
+    * The package index (*Packages* file) lists all available packages and their
+        metadata. This file is signed with the private key.
+    * A detached signature (*Packages.asc*) is generated and placed alongside the
+        *Packages* file.
+
+#. Verify Signature During Package Installation:
+
+    * When opkg installs a package, it verifies the signature of the *Packages* file
+        or the package itself using the public key.
+    * The public key is preloaded or added to the system's trusted keyring, such as
+        opkg-key (see :ref:`How keys are added to the opkg-keyring` for more details
+        on how this is accomplished).
+
+#. Trust Management:
+
+    * Trusted keys are stored in a keyring directory (e.g., /usr/share/opkg/keyrings/).
+    * During package installation, opkg ensures that the signature matches one of the
+        trusted keys.
+
+Workflow:
+---------
+Developer Workflow:
+
+    * Developers generate the *Packages* file in the feed and sign it with their private key.
+    * The packages and the index file (*Packages*) are uploaded to the package repository
+        along with the public key.
+
+Target System:
+
+    * The public key is pre-installed in the Base System Image or can be added to the
+        target system by installing/upgrading the opkg-keyrings IPK.
+    * When a package is installed, opkg checks the signature before proceeding.
+
+.. _How keys are added to the opkg-keyring:
+
+How keys are added to the opkg-keyring
+======================================
+The Base System Image includes the signing keys, typically located at /usr/share/opkg/keyrings,
+preloaded into the keyring.
+
+However, if the signing keys used by the feeds are rotated, the target system requires an update
+to add the new key and remove any obsolete keys no longer in use.
+
+The latest version of the opkg-keyrings package, available on the distribution feed
+(https://download.ni.com/#ni-linux-rt/feeds/dist/), is built with the updated signing keys.
+Upgrading this package on the target system installs the new signing key to
+/etc/share/opkg/keyrings, updates the keyring by adding the new key, and
+removes any deprecated keys no longer used to sign the feeds.
+
+By upgrading the opkg-keyrings package, the system ensures its keyring remains up-to-date
+and secure, seamlessly handling the addition of new keys and the removal of outdated ones.
+
+How to upgrade their keyring IPK via the command line, MAX, and SystemLink:
+===========================================================================
+Command Line:
+-------------
+There are two ways to add new keys to the opkg-keyring:
+
+Manually add with opkg-key:
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Download the new signing key (the public key):
+
+.. code-block:: bash
+
+    wget https://download.ni.com/#ni-linux-rt/feeds/dist/Packages.gz.asc
```

Comment:
Is this really the public key? My understanding is that this is just the signed data that would be different for files signed with same key.

The section makes it sound like you can download the public key from this file and add it to your system.. which I think is wrong.

Reply:
Yes, I was wrong about this. I had been following this when I had manually signed the package index file while testing. I have removed this now.